cpu/stm32/stmclk/stmclk_l4wb.c

```diff
@@ -57,46 +46,128 @@
  * @{
  */
 /* figure out which input to use */
-#if (CLOCK_HSE)
-#define PLL_IN                      CLOCK_HSE
+#if CONFIG_CLOCK_PLL_SRC_MSI
+#define PLL_SRC                     RCC_PLLCFGR_PLLSRC_MSI
+#elif CONFIG_CLOCK_PLL_SRC_HSE && CONFIG_BOARD_HAS_HSE
 #define PLL_SRC                     RCC_PLLCFGR_PLLSRC_HSE
 #else
-#define PLL_IN                      (48000000) /* MSI @ 48MHz */
-#define PLL_SRC                     RCC_PLLCFGR_PLLSRC_MSI
+#define PLL_SRC                     RCC_PLLCFGR_PLLSRC_HSI
 #endif
 
 /**check configuration and get the corresponding bitfields */
-#if (CLOCK_PLL_M < 1 || CLOCK_PLL_M > 8)
+#if (CONFIG_CLOCK_PLL_M < 1 || CONFIG_CLOCK_PLL_M > 8)
```

Comment:
When selecting MSI as the system clock source I get compile errors. These checks should be guarded right?

Reply:
In fact, they should unguarded from the default header file in `boards/common/stm32/include` because they must always be defined, even from clock modes that are not used. This is because now all branches of the clock configuration are built, even if unused and optimized by the compiler.

Reply:
It should be fixed now.